### PR TITLE
Effect.cancel(ids:) prefer array of ids and deprecate variadic overload

### PR DESCRIPTION
**Pull request by iampatbrown**
As discussed in #1009

Making this a draft so I can add more information tomorrow. Ran out of time. 

I had a quick look on GitHub and only found a couple of projects using the variadic overload.

### CODE REVIEW

File: Sources/ComposableArchitecture/Internal/Deprecations.swift [hunk 1]
```diff
@@ -3,6 +3,20 @@ import Combine
 import SwiftUI
 import XCTestDynamicOverlay
 
+// NB: Deprecated after 0.34.0:
+
+extension Effect {
+  @available(
+    *,
+    deprecated,
+    message: "For improved safety, using a variadic list is no longer supported. Use an array of identifiers instead. For more on this change, see: https://github.com/pointfreeco/swift-composable-architecture/pull/1041"
```
Comment (iampatbrown) [inline Sources/ComposableArchitecture/Internal/Deprecations.swift:12]: @stephencelis are you fine with this? The plan is to add a some context to the PR outlining the issue related to supporting both `AnyHashable...` and `[AnyHashable]`. Not sure if safety is the right word though.

Comment (stephencelis) [inline Sources/ComposableArchitecture/Internal/Deprecations.swift:12]:
```suggestion
    message: "Using a variadic list is no longer supported. Use an array of identifiers instead. For more on this change, see: https://github.com/pointfreeco/swift-composable-architecture/pull/1041"
```

**Review by stephencelis — APPROVED** [review]
Thanks!

File: Sources/ComposableArchitecture/Internal/Deprecations.swift (hunk 1 above)
Comment (iampatbrown) [reply Sources/ComposableArchitecture/Internal/Deprecations.swift]: Good change. I'll update the PR description tonight.

**Pull request merged**